### PR TITLE
EN-2412 DE component uninstallation

### PR DESCRIPTION
**Pull request by jeffgo10**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: src/api/digital-exchange/components.js [hunk 1]
```diff
@@ -36,7 +43,26 @@ export const getDEComponentInstall = id => (
   makeRequest({
     uri: `/api/digitalExchange/install/${id}`,
     method: METHODS.GET,
-    mockResponse: COMPONENT_INSTALLATION_IN_PROGRESS,
+    mockResponse: COMPONENT_INSTALLATION_COMPLETED,
```
Comment (gidesan) [inline src/api/digital-exchange/components.js]: why this change?

File: src/state/digital-exchange/components/reducer.js [hunk 2]
```diff
@@ -23,25 +25,32 @@ const selected = (state = {}, action = {}) => {
...
-      if (componentIndex === -1) {
-        return state;
-      }
-      const newState = state.slice();
-      newState.splice(componentIndex, 1, {
-        ...state[componentIndex],
-        installed: true,
-      });
-      return newState;
+      return findAndSetInstall(state, action.payload.id, true);
```
Comment (gidesan) [inline src/state/digital-exchange/components/reducer.js]: What about three methods? A `markComponentInstallStatus` with a boolean argument, plus, `markComponentAsInstalled` (calls markComponentInstallStatus with true) `markComponentAsUninstalled` (calls markComponentInstallStatus with true). Could be clearer for the reader.

File: src/api/digital-exchange/components.js (hunk 1 above)
Comment (jeffgo10) [reply src/api/digital-exchange/components.js]: this was changed because when using mock mode it doesn't really finish the mock installation process when you try to install. I can change it back though but I might as well add a comment to use `COMPONENT_INSTALLATION_COMPLETED` when testing installation in mock mode. would that be okay?

File: src/state/digital-exchange/components/reducer.js (hunk 2 above)
Comment (jeffgo10) [reply src/state/digital-exchange/components/reducer.js]: i'll see what i can do with this. thanks for the tip

**Comment by jeffgo10** [comment]
changes done. please review @gidesan

**Comment by entando-service** [comment]
All is well

File: src/api/digital-exchange/components.js (hunk 1 above)
Comment (gidesan) [reply src/api/digital-exchange/components.js]: ok then, keep your change

Comment (jeffgo10) [reply src/api/digital-exchange/components.js]: reverted

**Comment by entando-service** [comment]
All is well
